JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -0,0 +1,182 @@
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.key
+import com.example.jetnews.ui.article.ArticleScreen
+import com.example.jetnews.utils.WindowSize
+import com.example.jetnews.utils.getWindowSize
+
+/**
+ * Displays the Home route.
+ *
+ * Note: AAC ViewModels don't work with Compose Previews currently.
+ *
+ * @param homeViewModel ViewModel that handles the business logic of this screen
+ * @param openDrawer (event) request opening the app drawer
+ * @param scaffoldState (state) state for the [Scaffold] component on this screen
+ */
+@Composable
+fun HomeRoute(
+    homeViewModel: HomeViewModel,
+    openDrawer: () -> Unit,
+    scaffoldState: ScaffoldState = rememberScaffoldState()
+) {
+    // UiState of the HomeScreen
+    val uiState by homeViewModel.uiState.collectAsState()
+
+    HomeRoute(
+        uiState = uiState,
+        onToggleFavorite = { homeViewModel.toggleFavourite(it) },
+        onSelectPost = { homeViewModel.selectArticle(it) },
+        onRefreshPosts = { homeViewModel.refreshPosts() },
+        onErrorDismiss = { homeViewModel.errorShown(it) },
+        onInteractWithList = { homeViewModel.interactedWithList() },
+        onInteractWithDetail = { homeViewModel.interactedWithDetail(it) },
+        openDrawer = openDrawer,
+        scaffoldState = scaffoldState,
+    )
+}
+
+/**
+ * Displays the Home route.
+ *
+ * This composable is not coupled to any specific state management.
+ *
+ * @param uiState (state) the data to show on the screen
+ * @param onToggleFavorite (event) toggles favorite for a post
+ * @param onRefreshPosts (event) request a refresh of posts
+ * @param onErrorDismiss (event) error message was shown
+ * @param openDrawer (event) request opening the app drawer
+ * @param scaffoldState (state) state for the [Scaffold] component on this screen
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun HomeRoute(
+    uiState: HomeUiState,
+    onToggleFavorite: (String) -> Unit,
+    onSelectPost: (String) -> Unit,
+    onRefreshPosts: () -> Unit,
+    onErrorDismiss: (Long) -> Unit,
+    onInteractWithList: () -> Unit,
+    onInteractWithDetail: (String) -> Unit,
+    openDrawer: () -> Unit,
+    scaffoldState: ScaffoldState
+) {
+    // Construct the lazy list states for the list and the details outside of deciding which one to
+    // show.
+    // This allows the associated state to survive beyond that decision, and therefore we get to
+    // preserve the scroll throughout any changes to the content.
+    val homeListLazyListState = rememberLazyListState()
+    val articleDetailLazyListStates = when (uiState) {
+        is HomeUiState.HasPosts -> uiState.postsFeed.allPosts
+        is HomeUiState.NoPosts -> emptyList()
+    }.associate { post ->
+        key(post.id) {
+            post.id to rememberLazyListState()
+        }
+    }
+
+    BoxWithConstraints {
+        val useListDetail = when (getWindowSize(maxWidth)) {
```

Comment:
Not everyone is familiar with the list/detail terminology. Could we use something more explicit?

```suggestion
        val showArticleDetails = when (getWindowSize(maxWidth)) {
```

Reply:
`showArticleDetails` is a little confusing, since `showArticleDetails` could be `false` and we still show the article, I'll rename this to `showFeedAndArticleDetails` I think.

Reply:
Even better, I can inline this since it's only used directly below to create the `homeScreenType`.